drivers/spi/spi_ll_stm32.h

```diff
@@ -53,12 +56,30 @@ struct spi_stm32_data {
 #endif
 };
 
+#ifdef CONFIG_SPI_STM32_DMA
+static inline uint32_t ll_func_get_reg_addr(SPI_TypeDef *spi, uint32_t location)
+{
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	if (location == SPI_STM32_DMA_TX) {
+		/* use direct register location until the LL_SPI_DMA_GetTxRegAddr exists */
+		return (uint32_t)&(spi->TXDR);
+	}
+	/* use direct register location until the LL_SPI_DMA_GetRxRegAddr exists */
+	return (uint32_t)&(spi->RXDR);
+#else
+	ARG_UNUSED(location);
+	return (uint32_t)LL_SPI_DMA_GetRegAddr(spi);
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+}
+#endif /* CONFIG_SPI_STM32_DMA */
+
 static inline uint32_t ll_func_tx_is_empty(SPI_TypeDef *spi)
 {
 #if defined(CONFIG_SOC_SERIES_STM32MP1X) || \
-	defined(CONFIG_SOC_SERIES_STM32H7X) || \
 	defined(CONFIG_SOC_SERIES_STM32U5X)
 	return LL_SPI_IsActiveFlag_TXP(spi);
+#elif defined(CONFIG_SOC_SERIES_STM32H7X)
```

Comment:
As per our discussion, it was good to check TXP because space is available in FIFO, at least for Interrupt and Polling use cases. But it maybe necessary to differentiate DMA use case where it is necessary to check transfert complete after DMA is complete (DMA is complete means SPI is filled with DMA but there may be remaininf data in FIFO not yet send on bus)

Reply:
Yes. In case of DMA transfer, after  the DMA Tx is done it is necessary to check the SPI Tx before filling the next DMA packets or ending the transfer. Depending on the soc, this is possible by
- checking the LL_SPI_IsActiveFlag_TXC for stm32H7 or stm32U5
- a combination of LL_SPI_IsActiveFlag_TXE  and LL_SPI_IsActiveFlag_BSY

Reply:
The stm32H7 and stm32U5 (and stm32MP1)  have the same SPI IP.  The TXC flag is tested ito report the tx fifo empty status  as follows:
```
static inline uint32_t ll_func_tx_is_empty(SPI_TypeDef *spi)
{
#if defined(CONFIG_SOC_SERIES_STM32MP1X) || \
	defined(CONFIG_SOC_SERIES_STM32U5X) || \
	defined(CONFIG_SOC_SERIES_STM32H7X)
	return LL_SPI_IsActiveFlag_TXC(spi);
#else
	return LL_SPI_IsActiveFlag_TXE(spi);
#endif
}

```

Reply:
commit 14fa24c944c29cb00078c7aa1fa43b61979ef7fd